v2/service/module/tracerprovider/spanignore.go

```diff
@@ -0,0 +1,35 @@
+package tracerprovider
+
+import (
+	"context"
+	"slices"
+
+	"go.opentelemetry.io/otel/sdk/trace"
+)
+
+type SpanFilter struct {
+	// Next is the next SpanProcessor in the chain.
+	Next trace.SpanProcessor
+
+	// Ignore is the set of dropped span names.
+	Ignore []string
+}
+
+func (f SpanFilter) OnStart(parent context.Context, s trace.ReadWriteSpan) {
+	f.Next.OnStart(parent, s)
+}
+
+func (f SpanFilter) Shutdown(ctx context.Context) error {
+	return f.Next.Shutdown(ctx)
+}
+
+func (f SpanFilter) ForceFlush(ctx context.Context) error {
+	return f.Next.ForceFlush(ctx)
+}
+
+func (f SpanFilter) OnEnd(s trace.ReadOnlySpan) {
+	if slices.Contains(f.Ignore, s.Name()) {
+		return
+	}
+	f.Next.OnEnd(s)
+}
```

Comment:
Voisko tän toteuttaa upotuksella niin ei tarvis nuita "passthrough" funkkareita tehä:
```go
type SpanFilter struct {
	trace.SpanProcessor

	// Ignore is the set of dropped span names.
	Ignore []string
}

func (f SpanFilter) OnEnd(s trace.ReadOnlySpan) {
	if slices.Contains(f.Ignore, s.Name()) {
		return
	}
	f.SpanProcessor.OnEnd(s)
}
```

Reply:
SpanProcessor is just an interface and we need an implementation to pass data to next processor